aquacomputer_d5next.c

```diff
@@ -505,45 +513,45 @@ static int aqc_read_string(struct device *dev, enum hwmon_sensor_types type, u32
 static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr, int channel,
 		     long val)
 {
-	int ret, pwm_value;
+	int ret, pwm_value, temp_sensor;
 	struct aqc_data *priv = dev_get_drvdata(dev);
 
 	switch (type) {
 	case hwmon_pwm:
-		switch (attr) {
-		case hwmon_pwm_enable:
-			switch (priv->kind) {
-			case d5next:
-				if (val < 0 || val > 3)
-					return -EINVAL;
-				break;
-			case quadro:
-			case octo:
-				if (val < 0 || val > priv->num_fans + 3)
-					return -EINVAL;
-				/* check if the fan wants to follow itself, as this is not supported */
-				if (val == channel + 4)
-					return -EINVAL;
-				/* check if the fan we want to follow is currently following another one, this is not supported */
-				if (val > 3) {
-					ret = aqc_get_ctrl_val(priv, priv->fan_ctrl_offsets[val - 4], 8);
-					if (ret < 0)
-						return ret;
-					/* fan is following another one */
-					if (ret > 2)
+		if (priv->fan_ctrl_offsets) {
```

Comment:
This isn't needed? We can only write to what's visible, and this check is already present in `aqc_is_visible()` for the `hwmon_pwm` case.

Reply:
fixed in 61eea5b